### PR TITLE
Use ListView item's description text as aria-describedby

### PR DESCRIPTION
**Pull request by reidbarber**
Closes 

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 📝 Test Instructions:

Check that in an item with a description (`<Text slot="description" />`). the row's aria-describedby references the id of the element with the description.

### CODE REVIEW

**Review by snowystinger — APPROVED** [review]

File: packages/@react-spectrum/list/src/ListViewItem.tsx [hunk 1]
```diff
@@ -109,6 +109,12 @@ export function ListViewItem<T>(props: ListViewItemProps<T>) {
 
   let showCheckbox = state.selectionManager.selectionMode !== 'none' && state.selectionManager.selectionBehavior === 'toggle';
   let {visuallyHiddenProps} = useVisuallyHidden();
+  let descriptionId = useSlotId();
+  rowProps['role'] = 'row';
+  rowProps['aria-label'] = item.textValue,
+  rowProps['aria-selected'] = state.selectionManager.selectionMode !== 'none' ? isSelected : undefined,
+  rowProps['aria-rowindex'] = item.index + 1,
+  rowProps['aria-describedby'] = descriptionId;
```
Comment (LFDanLu) [inline packages/@react-spectrum/list/src/ListViewItem.tsx:117]: These should go into `useListItem` if possible (most are probably in there). The merge conflict must have been confusing haha sorry about that

Comment (reidbarber) [reply packages/@react-spectrum/list/src/ListViewItem.tsx:117]: Forgot about that, will do

File: packages/@react-aria/list/src/useListItem.ts [hunk 2]
```diff
@@ -166,12 +166,15 @@ export function useListItem<T>(props: AriaListItemOptions, state: ListState<T>,
     }
   };
 
+  let descriptionId = useSlotId();
```
Comment (LFDanLu) [inline packages/@react-aria/list/src/useListItem.ts:169]: You'll want to return this in a `descriptionProps` much like https://github.com/adobe/react-spectrum/blob/main/packages/%40react-aria/listbox/src/useOption.ts#L156-L161. I've tried doing this locally but having difficulties getting it to announce anything but the aria-label, changing it to aria-describedby + aria-labelledby doesn't seem to work that well either (still only announces the aria-labelledby. 

Will need to experiement which a labelling strategy that announces properly

Comment (LFDanLu) [reply packages/@react-aria/list/src/useListItem.ts:169]: Maybe a single aria-labelledby that has the ids of the content + description?

Comment (LFDanLu) [reply packages/@react-aria/list/src/useListItem.ts:169]: Also now that I look at it, not sure why we have a slot for content and for text?

Comment (reidbarber) [reply packages/@react-aria/list/src/useListItem.ts]: I'm seeing the same in VO, lets take a look later today and see what the best solution is

**Comment by snowystinger** [comment]
What do i need to do to get it to read the description in VO? or not quite ready yet?

**Comment by reidbarber** [comment]
Updated to use `aria-labelledby` when a description is provided, as suggested by @majornista. VO should now read off description after label

File: packages/@react-aria/list/src/useListItem.ts (hunk 2 above)
Comment (reidbarber) [reply packages/@react-aria/list/src/useListItem.ts]: Resolved, see https://github.com/adobe/react-spectrum/pull/3121#issuecomment-1138863077

File: packages/@react-spectrum/list/test/ListView.test.js [hunk 3]
```diff
@@ -592,6 +593,21 @@ describe('ListView', function () {
...
+        <Item textValue="Label">
+          <Text>Label</Text>
+          <Text slot="description">Description</Text>
+        </Item>
+      </ListView>
+    );
+
+    let rows = getAllByRole('row');
+    let description = within(rows[0]).getByText('Description');
+    expect(rows[0]).toHaveAttribute('aria-describedby', description.id);
```
Comment (snowystinger) [inline packages/@react-spectrum/list/test/ListView.test.js:608]: this should also check for labelledby now right?
does it need to have both?

**Review by snowystinger — COMMENTED** [review]
Is it just me or has it stopped announcing the title slot and now only does the description?
https://reactspectrum.blob.core.windows.net/reactspectrum/38cc771ff4379bbcb80a125c668a16a432af777a/storybook/index.html?path=/story/listview--long-text&providerSwitcher-toastPosition=bottom
Just tried it in Safari with VO

**Comment by reidbarber** [comment]
> Is it just me or has it stopped announcing the title slot and now only does the description? https://reactspectrum.blob.core.windows.net/reactspectrum/38cc771ff4379bbcb80a125c668a16a432af777a/storybook/index.html?path=/story/listview--long-text&providerSwitcher-toastPosition=bottom Just tried it in Safari with VO

@snowystinger I think with that story you linked, the textValue is 'row 2', and that does get read. Probably should just update that story with a better textValue.

Screen Shot 2022-05-31 at 5 09 56 PM

**Comment by reidbarber** [comment]
> Hmmm... Really? I see they are "row 1 with a very very very very" and "Text slot with a really really really" but all that VO says is "row 1" or "row 2" followed by the description Screen Shot 2022-05-31 at 3 59 59 PM

Yeah, the `<Text>` content won't be read because we're just using textValue as the `aria-label`:

https://github.com/adobe/react-spectrum/blob/3d340d00a9a45d60d9488234f583195843420f67/packages/%40react-aria/list/src/useListItem.ts#L167

and I think generally users would just set both to the same thing. Perhaps we could use one if the other isn't set. Not sure which would take precedence.

**Comment by snowystinger** [comment]
ooo, heh, that textValue, yeah, lets update that story so it doesn't confuse me in the future

**Review by snowystinger — APPROVED** [review]
This is working as I expect now in Safari with VO

**Review by LFDanLu — APPROVED** [review]
LGTM, NVDA announces the description and title only once as expected. Talkback is very verbose in its announcement but that is an issue noted before these changes and seems tied to grid announcement issues in Talkback in general (see "(Investigate) ListView issues" ticket in board for prior investigations on this)

**Pull request merged**